symfony/assets/scripts/class/navbar/Navbar.ts

```diff
@@ -138,7 +139,31 @@ class Navbar {
 
     this.separator = new NavbarElement(separator);
     this.prioritizedLinks = links
-      .map((node) => new NavbarLink(node, this.menu.cloneAppend(node)))
+      .map((node): NavbarLink => {
+        let ConstructorForNavbarLink: new (
+          node: HTMLAnchorElement,
+        ) => NavbarLink;
+
+        switch (node.dataset["navitemType"]) {
+          case "news": {
+            ConstructorForNavbarLink = NewsNavbarLink;
+            break;
+          }
+          case undefined: {
+            ConstructorForNavbarLink = NavbarLink;
+            break;
+          }
+          default: {
+            throw new Error(
+              `Unrecognized navbar link type '${node.dataset["navitemType"]}'`,
+            );
+          }
+        }
+
+        const navbarLink = new ConstructorForNavbarLink(node);
+        navbarLink.addToMenu(this.menu);
+        return navbarLink;
+      })
```

Comment:
Wouldn't it be easier to declare `let navbarLink: TypeHint` and construct it within the `switch`?